tools/osbuilder/Makefile

```diff
@@ -23,9 +23,9 @@ TARGET_IMAGE          := $(IMAGES_BUILD_DEST)/kata-containers.img
 TARGET_INITRD         := $(IMAGES_BUILD_DEST)/kata-containers-initrd.img
 
 VERSION_FILE   := ./VERSION
-VERSION        := $(shell grep -v ^\# $(VERSION_FILE))
+VERSION        := $(if $(shell test -f $(VERSION_FILE)),$(shell grep -v ^\# $(VERSION_FILE)),"unknown")
```

Comment:
What would be the case where the `VERSION` file is not existent / not present?

Our releases do include such file, and any tarball generated on GitHub would as well.

Reply:
I am not opinionated here. Once scenario *can* be where such release is being used to create packages (RPMs). For instance, a package that builds and ships the Kata host side components, or a minimal package that ships the relevant tooling/scripting (from osbuilder) to compose the guest components (rootfs, ...).

When creating such packages, my thinking is that one would merely want to include the minimal set of required files to compose the UVM.

Now, the VERSION file (which is symlinked to various places) seems to be less useful for some scenarios - for instance in the case where the UVM/rootfs is being assembled based on pre-built components (no build steps are executed, merely composing the UVM).

Looking at the code, the only real use of the VERSION file is within `create_summary_file` when we want to compose the rootfs based on pre-built components. Not everyone using kata-containers may be really interested in this summary file.

This is why I would propose to not 'hard depend' on this VERSION file and allow it to be inaccessible.

No emotional attachment here. If people think otherwise that's fair. :)

Note: Another topic may be in the future whether to include a summary file in the UVM at all (buzzword build reproducibility). :) Maybe an idea could be to produce this file alongside the rootfs artifact or make it an optional step.

Reply:
> For instance, a package that builds and ships the Kata host side components, or a minimal package that ships the relevant tooling/scripting (from osbuilder) to compose the guest components (rootfs, ...).

Oh, I see.  In this case we already installed such a tool in the system, and it'd still depend on the VERSION file to be placed somewhere, otherwise it'd bail.

I am not against having an "unknown" version here, as long as you're fine on having this on your side as well.

Ideally the release process should do better, and actually just have a version set there that the tool could use, without depending on the VERSION file, but this is orthogonal to your PR.

Reply:
I have aligned the read-in of the VERSION variable across all Makefiles